lib/core/src/chain_swap.rs

```diff
@@ -62,6 +90,87 @@ impl ChainSwapStateHandler {
         }
     }
 
+    pub(crate) async fn rescan_chain_swaps(&self) -> Result<()> {
+        let current_height = self.bitcoin_chain_service.lock().await.tip()?.height as u32;
+        let chain_swaps: Vec<ChainSwap> = self
+            .persister
+            .list_chain_swaps()?
+            .into_iter()
+            .filter(|s| s.direction == Direction::Incoming)
+            .collect();
+        info!(
+            "Rescanning {} Chain Swap(s) at height {}",
+            chain_swaps.len(),
+            current_height
+        );
+        for swap in chain_swaps {
+            if let Err(e) = self.rescan_chain_swap(&swap, current_height).await {
+                error!("Error rescanning Chain Swap {}: {e:?}", swap.id);
+            }
+        }
+        Ok(())
+    }
+
+    async fn get_confirmed_unspent_sat(&self, swap: &ChainSwap) -> Result<u64> {
+        let swap_script = swap.get_lockup_swap_script()?.as_bitcoin_script()?;
+        let script_pubkey = swap_script
+            .to_address(self.config.network.as_bitcoin_chain())
+            .map_err(|e| anyhow!("Error getting script address: {e:?}"))?
+            .script_pubkey();
+        Ok(self
+            .bitcoin_chain_service
+            .lock()
+            .await
+            .script_list_unspent(script_pubkey.as_script())?
+            .iter()
+            .filter(|u| u.height > 0)
+            .map(|u| u.value)
+            .sum())
+    }
+
+    async fn rescan_chain_swap(&self, swap: &ChainSwap, current_height: u32) -> Result<()> {
+        let monitoring_block_height =
+            swap.timeout_block_height + CHAIN_SWAP_MONTIORING_PERIOD_BITCOIN_BLOCKS;
+        let is_swap_expired = current_height > swap.timeout_block_height;
+        let is_monitoring_expired = current_height > monitoring_block_height;
+
+        match swap.state {
```

Comment:
Just as an attempt to simplify the match statement here.
Can we replace the match with an if of one condition that checks if the swap  is: 
`!is_monitoring_expired && expired || swap.state == Refundable`

If the given condition is true then update the new state (only if defer from the current state) according to the confirmed unspent amount.
positive => Refundable
zero => if claim then Complete else Failed